File: mobile/test/deadline_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:planner/deadline.dart';

void main() {
  final now = DateTime.parse('2026-08-20T12:00:00.000Z');

  test('deadline yaklaşan, geciken ve tamamlanan durumları ayırır', () {
    expect(deadlineState(null, false, now: now), DeadlineState.none);
    expect(
      deadlineState('2026-08-20T15:00:00.000Z', false, now: now),
      DeadlineState.soon,
    );
    expect(
      deadlineState('2026-08-22T15:00:00.000Z', false, now: now),
      DeadlineState.upcoming,
    );
    expect(
      deadlineState('2026-08-20T10:00:00.000Z', false, now: now),
      DeadlineState.overdue,
    );
    expect(
      deadlineState('2026-08-20T10:00:00.000Z', true, now: now),
      DeadlineState.completed,
    );
  });
}

File: web/src/lib/deadline.test.ts
import { describe, expect, it } from 'vitest';
import { deadlineFromInput, deadlineState, deadlineToInput } from './deadline.ts';

describe('card deadline', () => {
  const now = new Date('2026-08-20T12:00:00.000Z');

  it('yaklaşan, geciken ve tamamlanan durumları ayırır', () => {
    expect(deadlineState(null, false, now)).toBe('none');
    expect(deadlineState('2026-08-20T15:00:00.000Z', false, now)).toBe('soon');
    expect(deadlineState('2026-08-22T15:00:00.000Z', false, now)).toBe('upcoming');
    expect(deadlineState('2026-08-20T10:00:00.000Z', false, now)).toBe('overdue');
    expect(deadlineState('2026-08-20T10:00:00.000Z', true, now)).toBe('completed');
  });

  it('yerel form değeri ile UTC API değeri arasında dönüşür', () => {
    const local = '2026-08-20T18:30';
    const expected = new Date(2026, 7, 20, 18, 30).toISOString();
    expect(deadlineFromInput(local)).toBe(expected);
    expect(deadlineToInput(expected)).toBe(local);
    expect(deadlineFromInput('')).toBeNull();
  });
});
